lib/instrumentation/aws-sdk/v3/dynamodb.js

```diff
@@ -72,21 +72,66 @@ function dynamoMiddleware(shim, config, next, context) {
   }
 }
 
-const dynamoMiddlewareConfig = {
-  middleware: dynamoMiddleware,
-  init(shim) {
-    shim.setDatastore(shim.DYNAMODB)
-    return true
-  },
-  type: InstrumentationDescriptor.TYPE_DATASTORE,
-  config: {
-    name: 'NewRelicDynamoMiddleware',
-    step: 'initialize',
-    priority: 'high',
-    override: true
+/**
+ * Wraps the deserialize middleware step to add the
+ * cloud.resource_id segment attributes for the AWS command
+ *
+ * @param {Shim} shim
+ * @param {Object} config AWS command configuration
+ * @param {function} next next function in middleware chain
+ * @returns {function}
+ */
+function resourceIdMiddlerware(shim, config, next) {
+  return async function wrappedResourceIdMiddlerware(args) {
+    let region
+    try {
+      region = await config.region()
```

Comment:
you should be able to pull the region from the segment as it's getting set in the common middleware

Reply:
It looks like the dynamo middleware gets executed first and then the common middleware so I can't access `aws.region` attribute on segment in the dynamo middleware.

Reply:
you control the order but i guess the common middleware awaits the result which means it'll call any other custom middleware. This is fine, I still don't understand why this is an async function since looking at the source code it just pulls this from a config object